### DISCUSSION TITLE
lovyanGFX vs M5Unified vs Chimera

### DISCUSSION THREAD

**Discussion by mhaberler** [lovyan03/LovyanGFX#471] [General]
I'm trying to understand the relation of these repos and their relative functionality

my understanding so far is (please call BS if I'm wrong):

- M5Unified is a M5Stack-only subset of lovyanGFX (or Chimera? or both?)
- once a feature works in lovyanGFX and is relevant for M5 hardware, it will/might wind up M5Unified 
- Chimera started as a replacement and superset of the M5Stack library for additional non-M5 hardware
- Chimera is ESP32-only; lovyanGFX supports hardware well beyond that (some ARM, native, wasm)
- since lovyanGFX is a Chimera dependency: what does Chimera add beyond lovyanGFX? I see some driver support which is non-graphic and non-M5 like MPU9250; what am I missing in the bigger picture?

backdrop to my question:

We currently work on a flight instrument project based on M5 hardware and M5Unified
the current UI is Arduino/M5Unified based; next iteration will be LVGL + possibly Squareline
we'd prefer not to be exclusively locked to either M5 or for that matter ESP32 downstream, so while I think M5Unified is the best and most solid basis it would would imply an M5+espressif lock-in
I am wary of the espressif lock-in as I have doubts about the long-term future of this architecture and the tooling is sub-par 
but for now it's the best price/performance and unfortunately lacks competition 
same goes for "OS" - I am lukewarm on both Arduino and IDF; ideally I'd use Zephyr but that seems like a non-starter for espressif
eventually 

I am trying to figure how the moving parts fit together
my current approach would be to use LVGL + lovyanGFX on M5 hardware
that would keep the route open to non-M5 hardware "below" and eventually Linux fb/ or even web with wasm

I'd be interested how people think about this puzzle

Michael

**Comment by tobozo** [comment]
> M5Unified is a M5Stack-only subset of lovyanGFX (or Chimera? or both?)M5Unfiied 

M5Unified is a depends of M5GFX which is a subset of LovyanGFX

Chimera-Core only demonstrated that unification is possible, it differs from M5Unified by role and quality (M5Unified code is much better), but it is brand agnostic.

Moreover, M5Unified lets the user create a single binary that works on any M5 device, while Chimera-Core uses macros to detect the boards and can only produce binaries that focus at one device at a time.

I use Chimera-Core whenever I need to capture the display, the ScreenShot service can export to images or video.

Other than that Chimera-Core brings no real added value to LovyanGFX unless you own one of the listed boards and feel too lazy to write your own LGFX Config.

**Comment by mhaberler** [comment]
thanks - I really appreciate the clarification!
